app/controllers/api_controller.rb

```diff
@@ -31,12 +31,40 @@ def current_facility_group
     current_user.facility.facility_group
   end
 
+  def current_sync_region
+    # This method selectively permits only FacilityGroup sync (via facility group ID)
+    # and block-level sync (via regions) and offers facility group as a safe fallback.
+    # Over time, the facility group ID support can be dropped and this method can
+    # allow other region types as well
+    return current_facility_group if requested_sync_region_id.blank?
+    return current_facility_group if requested_sync_region_id == current_facility_group.id
+    return current_block if block_level_sync?
+
+    current_facility_group
+  end
+
+  def block_level_sync?
+    current_user.block_level_sync? && requested_sync_region_id == current_block.id
+  end
+
+  def current_block
+    # Fetching current block from current_facility is safer
+    # than fetching it by Region.find(requested_sync_region_id)
+    # since the requested_sync_region_id can be an FG id.
+    # This can be replaced in the future when facility group ID support is dropped.
+    current_facility.region.block_region
```

Comment:
I'd posit we stick to getting the block from the current facility permanently. Regardless of what the user is asking to sync, the current facility (which comes from the X-Facility-ID header, right?) is where the user is.

Reply:
Are we still getting a new header for the `region_type` as well as the `region_id` ? I thought I saw that in earlier discussions.

If so that could allow us to be safer when region sync is enabled as we could constraint the lookup by both:

```
current_user.accessible_regions.where(id: requested_sync_region_id, region_type: requested_sync_region_type)
```

Reply:
We aren't getting a separate header for `region_type`. In fact this was one of the reasons we went with `Region`, since it allows us to drop this header 😄 .